symphony-utility/src/main/java/org/symphonyoss/symphony/bots/utility/client/SymphonyClientUtil.java

```diff
@@ -17,16 +26,50 @@
 public class SymphonyClientUtil {
 
   private static final int DEFAULT_MAX_MESSAGES = 10;
+  private static final String MESSAGE_COULD_NOT_CREATE_FILE = "Couldn't create a file.";
+  private static final String MESSAGE_ATTACHMENT_NOT_FOUND = "Attachment not found.";
+  private static final String MESSAGE_FAILED_TO_CREATE_FILE = "Failed to create File";
 
   private SymphonyClient symphonyClient;
 
   public SymphonyClientUtil(SymphonyClient symphonyClient) {
     this.symphonyClient = symphonyClient;
   }
 
+  public File getFileAttachment(SymAttachmentInfo symAttachmentInfo, SymMessage symMessage) {
+    String tmpDir = System.getProperty("java.io.tmpdir");
+    File directory = new File(tmpDir + File.separator + symAttachmentInfo.getId());
+    if (!directory.exists()) {
+      directory.mkdir();
+    }
+
+    File file = new File(directory + File.separator + symAttachmentInfo.getName());
+    try {
+      file.createNewFile();
+    } catch (IOException e) {
+      throw new BadRequestException(MESSAGE_COULD_NOT_CREATE_FILE);
+    }
+
+    byte[] aByte;
+    try {
+      aByte =
+          symphonyClient.getAttachmentsClient().getAttachmentData(symAttachmentInfo, symMessage);
+    } catch (AttachmentsException e) {
+      throw new BadRequestException(MESSAGE_ATTACHMENT_NOT_FOUND);
+    }
+
+    InputStream inputStream = new ByteArrayInputStream(aByte);
+    try {
+      FileUtils.copyInputStreamToFile(inputStream, file);
+    } catch (IOException e) {
+      throw new BadRequestException(MESSAGE_FAILED_TO_CREATE_FILE);
```

Comment:
This is a server error, so you should change this exception to InternalServerErrorException